bci_tester/data.py

```diff
@@ -220,15 +220,22 @@ def create_container_version_mark(
 
 
 if BCI_DEVEL_REPO is None:
-    # from SLE 15 SP6 onward we use the unauthenticated CDN
-    cdn_prefix = "installer-updates" if OS_SP_VERSION >= 6 else "updates"
-    BCI_DEVEL_REPO = f"https://{cdn_prefix}.suse.com/SUSE/Products/SLE-BCI/{OS_MAJOR_VERSION}-SP{OS_SP_VERSION}/{LOCALHOST.system_info.arch}/product/"
+    if OS_VERSION == "tumbleweed":
+        BCI_DEVEL_REPO = "http://download.opensuse.org/tumbleweed/repo/oss/"
+    else:
+        # from SLE 15 SP6 onward we use the unauthenticated CDN
+        cdn_prefix = "installer-updates" if OS_SP_VERSION >= 6 else "updates"
+        BCI_DEVEL_REPO = f"https://{cdn_prefix}.suse.com/SUSE/Products/SLE-BCI/{OS_MAJOR_VERSION}-SP{OS_SP_VERSION}/{LOCALHOST.system_info.arch}/product/"
     _BCI_REPLACE_REPO_CONTAINERFILE = ""
 else:
     bci_repo_path = f"/etc/zypp/repos.d/{BCI_REPO_NAME}.repo"
-    # only try to replace the baseurl if the file exists and we have perl
-    # available (should be more ubiquitous than sed)
-    _BCI_REPLACE_REPO_CONTAINERFILE = f"RUN if [ -e {bci_repo_path} ] && [ -e /usr/bin/perl ]; then perl -pi -e 's|baseurl.*|baseurl={BCI_DEVEL_REPO}|g' {bci_repo_path}; fi"
+    bci_repo_replace = "${line/baseurl*/baseurl = " + BCI_DEVEL_REPO + "}"
+    _BCI_REPLACE_REPO_CONTAINERFILE = f"""RUN if [ -e {bci_repo_path} ]; then \\
```

Comment:
I don't get why we allow for multiple repositories here, but assert that there is only one in `test_all.py` below.

Reply:
There are always multiple repos. In test_all we are testing that we can find exactly one with the expected name because that's the one we're going to modify